From 0ddc34c88813f514982f439056824b4b5972ff57 Mon Sep 17 00:00:00 2001
From: Jordan Williams <jordan@jwillikers.com>
Date: Tue, 23 Apr 2024 07:43:00 -0500
Subject: [PATCH 20/21] meson: Add build-enosys option

Add a check to ensure that the linux/audit.h header exists for enosys.

Signed-off-by: Jordan Williams <jordan@jwillikers.com>
(cherry picked from commit de62df14e1fb94302a330bf18b4ff9f89a32b7fa)
---
 meson.build       | 29 ++++++++++++++++-------------
 meson_options.txt |  2 ++
 2 files changed, 18 insertions(+), 13 deletions(-)

diff --git a/meson.build b/meson.build
index 8ba03ca760..55e1ecde6e 100644
--- a/meson.build
+++ b/meson.build
@@ -3106,19 +3106,22 @@ syscalls_h = custom_target('syscalls.h',
              cc.cmd_array(), get_option('c_args')],
 )
 
-if cc.compiles(fs.read('include/audit-arch.h'), name : 'has AUDIT_ARCH_NATIVE')
-  exe = executable(
-    'enosys',
-    'misc-utils/enosys.c', syscalls_h,
-    include_directories : includes,
-    link_with : [lib_common],
-    install_dir : usrbin_exec_dir,
-    install : true)
-  if not is_disabler(exe)
-    exes += exe
-    manadocs += ['misc-utils/enosys.1.adoc']
-    bashcompletions += ['enosys']
-  endif
+have_linux_audit_h = cc.has_header('linux/audit.h')
+have_audit_arch_native = cc.compiles(fs.read('include/audit-arch.h'), name : 'has AUDIT_ARCH_NATIVE')
+
+opt = get_option('build-enosys').require(have_linux_audit_h and have_audit_arch_native).allowed()
+exe = executable(
+  'enosys',
+  'misc-utils/enosys.c', syscalls_h,
+  include_directories : includes,
+  link_with : [lib_common],
+  install_dir : usrbin_exec_dir,
+  install : opt,
+  build_by_default : opt)
+if opt and not is_disabler(exe)
+  exes += exe
+  manadocs += ['misc-utils/enosys.1.adoc']
+  bashcompletions += ['enosys']
 endif
 
 exe = executable(
diff --git a/meson_options.txt b/meson_options.txt
index 1850d1d174..9f09f3220c 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -203,6 +203,8 @@ option('build-pipesz', type : 'feature',
        description : 'build pipesz')
 option('build-fadvise', type : 'feature',
        description : 'build fadvise')
+option('build-enosys', type : 'feature',
+       description : 'build enosys')
 option('build-setterm', type : 'feature',
        description : 'build setterm')
 option('build-schedutils', type : 'feature',
-- 
2.44.0

